sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/logicaltypes/OneOfType.java

```diff
@@ -103,12 +100,12 @@ public String getIdentifier() {
 
   @Override
   public FieldType getArgumentType() {
-    return FieldType.BYTES;
+    return FieldType.row(oneOfSchema);
   }
 
   @Override
-  public byte[] getArgument() {
```

Comment:
this looks like a breaking change. However I am not sure the implication, good to involve others reviewer, possibly @robertwb ?

Reply:
The previous behavior seems incorrect as proto serialization is not necessarily deterministic and the serialized proto equality doesn't match schema equality logic.

But I think that you are correct that this would break schema update compatability because LogicalType base class appears to be serializable.

@robertwb Any ideas on how to best proceed? We could
1. keep OneOf as-is and just document that it's equality isn't guaranteed to be correct due to proto serialization and clear the uuid before serializing the proto.
2. add a new type with correct semantics and deprecate this one
3. try to integrate w/ update compatibility version but that seems difficult given schemas are created all over.  We coudl have some static method we call that modifies static state instead of plumbing the option everywhere.  getArgument base is to return an Object so we can have single class support both types based upon the configuration.

Reply:
Good question. 

First, let's separate this change out into its own PR, so we can get the rest (which looks good) in. 

I'm worried about what existing pipelines might break if we change this. Does a (null?) Row eventually get serialized with its schema as proto bytes anyway? We could look into explicitly using deterministic proto serialization here as an improvement at least.

Reply:
Some change of OneOf is needed because it's schema now has a uuid set and thus doesn't compare equal.

However I changed to keep using serialized proto but to recursively remove the UUID from the schema proto before serializing.

PTAL